### ISSUE TITLE
Phase 2 Roadmap: Algorithms & Models 

### ISSUE THREAD

**Issue by mdancho84** [business-science/modeltime#5]
These algorithms on on the roadmap for Phase 2. 

## Structural Models

- [ ] BSTS

## Forecast

- [ ] STLM - Decomposition Models
- [ ] TBATS 
- [ ] BATS
- [ ] NNETAR

**Comment by edgBR** [comment]
Dear @mdancho84,

I see a lot of parallelism between model time and the fable package. Even that fable is designed for multiple time series models I believe that sharing some of the structures could be very beneficial for both projects. I have noticed that for instance you have a model time table. Would it be possible that you look into the mable (mdl_df) class from fable?

BR
/E

**Comment by mdancho84** [comment]
Yes, I agree there are parallels. I'm not sure what specifically to look at compared to `fable`. I do plan to incorporate most of the `forecast` package functionality into `modeltime`. Scalable time series is something else I will look into down the road, but it becomes more complex when tuning is involved.

**Comment by edgBR** [comment]
Hi @mdancho84 

I think the main issue with tunning is that you will have a conflict when you are running fable in parallel as you will not have control over the forked processes that the HPO needs. However I am not an expert in the topic but I for sure understand the challenges here.

However I see now 3 main packages to perform forecasting in R:

forecastML: for direct forecasting of multiple grouped time series using machine learning and deep learning.
fable: for recursive forecasting using time series models and prophet
modeltime: for recursive forecasting using machine learning and deep learning.

forecastML plans to be compatible with fable and there is even a vignette in order to explain reconciliation forecasts.

I think that it would be amazing if all of this packages become part of a modern time series forecasting ecosystem.

BR
/E

**Comment by mdancho84** [comment]
I’m a big fan of fable, I just couldn’t get it to work for anything using machine learning and deep learning.

The tidymodels system is an amazing evolution of caret that is focused on integration and standardization of machine learning. This makes it much easier to control and simplify the vast ML & DL ecosystem. This is why it only takes minimal lines of code to go from idea to execution. Modeltime and timetk simply add a workflow fore forecasting on top of the excellent tidymodels ecosystem. 

I dont know if there will be one framework that will consolidate all of these great ideas. I agree it would be amazing. 

My goal is to first and foremost help those that want to create high performance forecasts have the tools available to do so in as few lines of code as possible.

**Comment by mdancho84** [comment]
TBATS is added. 

```
seasonal_reg(
    seasonal_period_1 = 7,
    seasonal_period_2 = 30,
    seasonal_period_3 = 365
) %>%
    set_engine("tbats")
```

**Comment by mdancho84** [comment]
## Deep Learning (RNN / LSTM) with Modeltime GluonTS

The `modeltime.gluonts` library has been created to handle Deep Learning by integrating the Time Series Deep Learning Library, GluonTS.  Resources:

- **Modeltime GluonTS Docs:** https://business-science.github.io/modeltime.gluonts/
- **GluonTS Docs:** https://ts.gluon.ai/index.html

**Comment by mdancho84** [comment]
> I don't recall if holtwinters works with modeltime, does it?

Yes, Holt-Winters is a special form of ETS. So you can use `exp_smoothing()` to make a holt winters model.

**Comment by mdancho84** [comment]
Added Conformal Interval Evaluation #173
